pkg/ccl/backupccl/restore_online.go

```diff
@@ -519,13 +504,35 @@ func (r *restoreResumer) cleanupAfterDownload(
 	ctx, sp := tracing.ChildSpan(ctx, "backupccl.cleanupAfterDownload")
 	defer sp.Finish()
 
-	executor := r.execCfg.InternalDB.Executor()
-
-	// Re-enable automatic stats collection on restored tables.
+	// Try to restore automatic stats collection preference on each restored
+	// table.
 	for _, table := range details.TableDescs {
-		_, err := executor.Exec(ctx, "enable-stats", nil, `ALTER TABLE $1 SET (sql_stats_automatic_collection_enabled = true);`, table.Name)
-		if err != nil {
-			log.Warningf(ctx, "could not enable automatic stats on table %s", table)
+		if err := sql.DescsTxn(ctx, r.execCfg, func(
+			ctx context.Context, txn isql.Txn, descsCol *descs.Collection,
+		) error {
+			b := txn.KV().NewBatch()
+			newTableDesc, err := descsCol.MutableByID(txn.KV()).Table(ctx, table.ID)
+			if err != nil {
+				return err
+			}
+			// Setting the AutoStatsSetting field to nil implies the table will have
+			// the detault auto stats settings.
+			newTableDesc.AutoStatsSettings = nil
+			if enabled, ok := details.PostDownloadTableAutoStatsSettings[uint32(table.ID)]; ok {
+				newTableDesc.AutoStatsSettings = &catpb.AutoStatsSettings{Enabled: &enabled}
```

Comment:
There are more settings in this proto that may be getting overwritten

Reply:
oof. good point. fixed.